loopchain/blockchain/blockchain.py

```diff
@@ -137,11 +139,22 @@ def _keep_order_in_penalty(self) -> bool:
         utils.logger.debug(f"_keep_order_in_penalty() : keep_order = {keep_order}")
         return keep_order
 
-    def reset_leader_made_block_count(self, is_switched_role: bool = False):
+    def reset_leader_made_block_count(self, need_check_switched_role: bool = False):
         """Clear all made_block_counter
 
         :return:
         """
+        if need_check_switched_role:
+            if self.__last_block.header.prep_changed_reason == NextRepsChangeReason.NoChange:
+                utils.logger.debug(f"There is no change in reps.")
+                return
+
+            new_reps = self.find_preps_addresses_by_roothash(self.__last_block.header.revealed_next_reps_hash)
+            new_node_type = NodeType.CommunityNode if ChannelProperty().peer_address in new_reps else NodeType.CitizenNode
+            is_switched_role = new_node_type != ChannelProperty().node_type
+        else:
+            is_switched_role = False
+
```

Comment:
I can see those lines has been migrated partially from `PeerManager.reset_all_peers`.
But I cannot understand what those lines do.

Please consider:
- Restore signature of `reset_leader_made_block_count` and its contents. 
- Extract those lines as `is_role_switched` or something reasonable, then pass its return value to `reset_leader_made_block_count`. (I think this method does more than one thing, which makes me confused.)